rl4co/models/nn/ops.py

```diff
@@ -12,26 +13,40 @@ def forward(self, x):
         return x + self.module(x)
 
 
+class AdaptiveSequential(nn.Sequential):
+    def forward(
+        self, *inputs: Union[Tuple[torch.Tensor], torch.Tensor]
+    ) -> Union[Tuple[torch.Tensor], torch.Tensor]:
+        for module in self._modules.values():
+            if type(inputs) == tuple:
+                inputs = module(*inputs)
+            else:
+                inputs = module(inputs)
+        return inputs
+
+
 class Normalization(nn.Module):
     def __init__(self, embed_dim, normalization="batch"):
         super(Normalization, self).__init__()
+        if normalization != "layer":
+            normalizer_class = {
+                "batch": nn.BatchNorm1d,
+                "instance": nn.InstanceNorm1d,
+            }.get(normalization, None)
 
-        normalizer_class = {"batch": nn.BatchNorm1d, "instance": nn.InstanceNorm1d}.get(
-            normalization, None
-        )
-
-        self.normalizer = normalizer_class(embed_dim, affine=True)
-
-    def init_parameters(self):
-        for name, param in self.named_parameters():
-            stdv = 1.0 / math.sqrt(param.size(-1))
-            param.data.uniform_(-stdv, stdv)
+            self.normalizer = normalizer_class(embed_dim, affine=True)
+        else:
+            self.normalizer = "layer"
 
     def forward(self, x):
         if isinstance(self.normalizer, nn.BatchNorm1d):
             return self.normalizer(x.view(-1, x.size(-1))).view(*x.size())
         elif isinstance(self.normalizer, nn.InstanceNorm1d):
             return self.normalizer(x.permute(0, 2, 1)).permute(0, 2, 1)
+        elif self.normalizer == "layer":
+            return (x - x.mean((1, 2)).view(-1, 1, 1)) / torch.sqrt(
+                x.var((1, 2)).view(-1, 1, 1) + 1e-05
+            )
```

Comment:
I am curious about the difference between this part and [`torch.nn.LayerNorm`](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm). 🤔